### PR TITLE
fix: go-ipfs 0.5 sync

### PR DESCRIPTION
**Pull request by hugomrdias**
- fix remote node ipfs http client path
- fix rabin params go-ipfs has validation and returns errors
- fix use cid instead of hash in the pin tests

### CODE REVIEW

File: test/files.js [hunk 1]
```diff
@@ -354,7 +354,7 @@ describe('files', function () {
 
     it('rabin chunker small chunks', () => {
       const options = {
-        chunker: 'rabin-16-16-16',
+        chunker: 'rabin-16-32-64',
```
Comment (achingbrain) [inline test/files.js:357]: Why was this change necessary?  `rabin-16-16-16` seems to be supported by js and go:

```console
$ ipfs add --chunker=rabin-16-16-16 ./package.json 
added QmTUSudDKFYaVHLdU1wRoHYXkydQaDch4Kt8CwuVFShpiU package.json
$ jsipfs add --chunker=rabin-16-16-16 ./package.json 
added QmTUSudDKFYaVHLdU1wRoHYXkydQaDch4Kt8CwuVFShpiU package.json
```

Comment (hugomrdias) [reply test/files.js:357]: api returns validation errors, min needs to bigger than avg, avg need to be smaller than max etc

Comment (achingbrain) [reply test/files.js:357]: Which api?  The js-IPFS api?  Should probably fix that.

Comment (hugomrdias) [reply test/files.js:357]: Didn't confirm, but I'm assuming go http api we didn't change anything.
I will look into it

Comment (achingbrain) [reply test/files.js:357]: Oh, no - I think you're correct, it's go-IPFS that's changed, I think I was running an old version just now (still 0.5.x but an old version from go-ipfs master):

```console
ipfs add --chunker=rabin-16-16-16 ./package.json 
Error: incorrect format: rabin-min must be smaller than rabin-avg
```

Comment (achingbrain) [reply test/files.js:357]: @Stebalien was this change intentional?

Comment (Stebalien) [reply test/files.js:357]: This chunker config is equivalent to a fixed-sized chunker. I'm not sure if this was intentional, but I'd like to leave it as it helps detect misuse of the chunker.

Comment (hugomrdias) [reply test/files.js:357]: added this item here https://github.com/ipfs/js-ipfs/issues/3030 so we replicate in js-ipfs
